### DISCUSSION TITLE
Tweak settings for better color on northern lights and the night sky

### DISCUSSION THREAD

**Discussion by gigaklatch** [aaronwmorris/indi-allsky#1450] [Q&A]
So, last night, right in the middle of the Perseid meteor shower some northern lights appeared.
![ccd1_20240812_015326](https://github.com/user-attachments/assets/c9b54c54-684d-4610-a5dd-cf34ede603a3)
![ccd1_20240812_023933](https://github.com/user-attachments/assets/d26cf34b-ef1e-4809-9051-fa108f397b6a)
I am glad that my camera caught as well as it did, however, I am a bit bummed by the lack of color. I took some photos with both my phone and my DSLR and in the pictures taken with those devices the aurora had som amazingly deep colors, yellow, green, pink, red, blue.

How should I change the settings on my allsky to allow it to display colors like that as well? Should I lower the exposures and stretching, the northern lights might be clipping to white? 

There are some color to the milky way and the stars, but they are quite bland. 

Please help

Kind regards

**Comment by aaronwmorris** [comment]
Do you have a detection mask setup?  That would be the first thing I would try.  This would allow the exposure to be balanced to more areas of the sky (including the edges).

Second, you may try boosting your saturation somewhere to 1.25 to 1.5.  This can really help with your star colors as well.

**Reply by gigaklatch** [reply]
So, I made a detection mask and used it during the following night. Luckily, the aurora did return. The result can be seen below:
![ccd1_20240813_003740](https://github.com/user-attachments/assets/974d684a-f1ad-439c-9c6f-a8be72e00647)
Here is a photo, straight out of my Nikon D750, also a 30 sec exposure, depicting the exact same moment for color reference:
![DSC_4276](https://github.com/user-attachments/assets/1998f0f4-0a84-4d99-b7d9-2c51b89636ec)
This camera was aimed at a patch of the sky corresponding to the bottom centre portion of the allsky image.

I did have the foresight to have save FITS enabled at the Allsky camera so here follows a link to a google drive folder containing the fits for the very same image shown first in this post:
[https://drive.google.com/drive/folders/1L-tK0v5m-WW4sWBm8lPziqcONn-ONSpd?usp=sharing](url)

Playing around with the FITS-processing tool in INDI-AllSky I can not for the life of me get the colors anywhere near the ones from my DSLR. And the portion of it closest to the horizon stays blown out.

Another thing I tried was opening the FITS-file in ASI Fits viewer, and the colors are still off, but seemingly not as overexposed:
![ccd1_20240813_003740_ASI](https://github.com/user-attachments/assets/21eb0709-7a7b-4319-ab88-d3f5a8b5cad5)

**Reply by aaronwmorris** [reply]
Can you post a view of your detection mask?

**Reply by gigaklatch** [reply]
Certainly,

![mask_base](https://github.com/user-attachments/assets/fce878b7-d6db-4066-a557-2bb5d45bd29a)

**Reply by aaronwmorris** [reply]
Thanks!

**Comment by aaronwmorris** [comment]
I was able to import the FITS file into on of my lab systems and play with the settings in the FITS Processor.  There are a couple of things going on here...

1.  White Balance.  Your DSLR is going to do a much better job at this because its software/hardware is designed around the frequency response of the sensor it was designed around.  indi-allsky does not really perform any type of White Balancing.  The software option for white balance really just tries to adjust the average (the peak of the histogram) of each color to more closely align.... which is basically "make things more white"  :-)
2. Even with the detection mask, the area of interest is everything within the mask.  Its best that the subject occupies the majority of view, but when things are mostly at the edge.... the calculation is still done on the whole area of the mask.  This is not always what you would want to happen.
3. The stretching algorithm did not really do any favors to the colors.  In fact, the stretching kind of washed things out and pushed the color toward "white".  Disabling stretching resulted in a better result for the aurora.  Actually, doing a more aggressive stretch was a bit better, setting the deviations value below 2 was a little better.  It is unfortunate, because the stretching really brings out things like the milky way.
4. Boosting the saturation helps a little bit with all sky cameras... a setting of 1.25 really helps the colors pop a bit more.

I wish I had better algorithms for white balancing.  Dealing with RAW data, it is very limiting to not to be available.

**Reply by gigaklatch** [reply]
1. I see. So it sounds like that there might be some usable data in those raw fits files, and that I could possibly white balance better and eke out some nicer colors using more powerful editing software?
2. I understand.
3. Yes, I noticed that too. Earlier I was not really using the stretching, until I notice I could bring out the milky way quite nicely, as seen in the image from the first nights with aurora in the starting post of this thread.
4.  I did find that to be somewhat good for bringing out some colors. It also seemed to bright up the light levels in the brightest areas and  make more pixels clip.

I really hope I can salvage these images and produces a nice timelapse of the aurora. 

5. I have seen much better results from other peoples all sky cameras? Could I perhaps reach better results with a different camera? I now have an SVBony SV705C which has the IMX585 sensor. Quite a large sensor and I have also found quite a large lens to use with it. Was hoping that that combination would be better than the Raspberry Pi HQ camera and cheap lens that I was using before.
 Do I need to turn to different software, such as AllskEye on a PC?

**Reply by aaronwmorris** [reply]
There is another aspect here with regard to the performance of the DSLR which makes this a bit of an unfair comparison.

The DSLR probably has at least an APS-C sized sensor with a lens that has better optics, a much greater aperture (~5-10mm), and is probably a bit slower in F-stop, and a more limited FoV.

All sky lenses are usually very fast, like f/2.0-2.5, tiny aperture (~1mm), and a huge FoV.  The sensors are also relatively tiny.  All of these aspects severely limit the results.

The HQ camera (via libcamera) can return some impressive results due to the AWB features native to libcamera.  However, there are some side effects to using the AWB (if you enable it), namely, it requires multiple internal exposures to produce a pretty picture.  If you take a 30s exposure, it might take 90-120s to produce the 30s exposure, which limits the number of pictures you take at night.  There is no free lunch.

**Reply by aaronwmorris** [reply]
As for the SV705c, that camera should be great.  However, for color fidelity, I do not think the sensor has an IR-Cut filter which can be a challenge for white balance.

Getting colors correct with astronomy cameras has been the subject of many flame wars.  :-)

I would welcome the comparison with AllSkEye.

**Reply by gigaklatch** [reply]
Yeah, the DSLR even has a full frame sensor.  So the area of the sensor is around 12 times greater, the area of each pixel about 4 times that of the IMX585. The lens used was a Samyang 14 mm lens stopped down to F/4. Not fair at all. I do not expect to get colors to that degree, but at least had hoped to achieve something at least a little bit reminiscent.

You are correct that the SV705C does not have any IR-CUT filter. I believe, also, that the IMX585 sensor is very sensitive to IR-light:
![6380591249619539312970518](https://github.com/user-attachments/assets/94be8063-ad15-493c-9dc5-4efaf27d0a9c)

The lens used on the SV705C is found here:
 [https://www.aliexpress.com/item/4000905611143.html?spm=a2g0o.productlist.main.11.e9dfim5lim5l5j&algo_pvid=4c8297ac-30c7-493a-9c21-878d429ee863&algo_exp_id=4c8297ac-30c7-493a-9c21-878d429ee863-5&pdp_npi=4%40dis%21SEK%21710.51%21710.51%21%21%2166.11%2166.11%21%40211b81b117237416327808284eb37b%2110000010484019276%21sea%21SE%214220988501%21X&curPageLogUid=6I6U3ggTjRje&utparam-url=scene%3Asearch%7Cquery_from%3A#nav-specification](url)

It is stated that the lens has IR-correction, not entirely sure what that means. It is unlikely that I can get any answer about that from the seller, I have not had much luck with sellers on Aliexpress, they seem to know nothing about the things that they have on offer.

Whatever that IR-correction does maybe I could look into having an IR-cut filter placed between the back of the lens and the protective window of the camera. I do not believe that there is a lot of space between those two surfaces, so it would have to be quite a thin filter.

One of the cheaper lenses I tried for my Pi HQ camera had a square piece of IR-cut filter crudely glued to the back of it.

**Reply by aaronwmorris** [reply]
According to this page:  https://www.svbony.com/blog/parameters-comparison-of-sv305-to-sv705c-cameras/

The SV705c just has the AR Anti-Reflective window.

**Reply by gigaklatch** [reply]
Then we all agree on that point.

**Comment by gigaklatch** [comment]
One thing I did note during the Aurora was that. I have a couple of Wyze Cam V3 pointed where the Aurora did show up. Using the color mode on those one could clearly see there was something there, a bit of pink and green cloudiness. When switching the cameras over to their IR-modes, where the IR-cut filter  is removed, the Aurora was even more clearly visible, though no longer with any color to them, as the IR mode switchess to black and white. The Aurora showing up more clearly suggests that there is a lot of IR light coming off of it. And with all of the pixels on the SV705C being very sensitive to IR it would mean that all the detects those photons, R, G as well as B, with no discrimination towards any particular color, hence the extreme brightness on my all sky images.

A somewhat cryptic and difficult to understand reply from the seller of my lens on Aliexpress suggests that the IR-correction helps in bringing the IR-wavelengths into focus together with the visible spectrum light. The lens seem to be meant for surveillance application, so it would not make much sense to have IR blocked.

Given all of the above, it seems to me that it would make sense to try and add an IR-cut filter between the camera and lens somehow. Like I mentioned earlier, there is not a lot of room to do so, so it might not be possible, but maybe it can be done.

**Reply by aaronwmorris** [reply]
Consider the option of replacing the window glass.  ZWO sells replacement windows for their cameras, so it gives you the option to swap between AR and IR options.  There might be a window option that could work for your camera.

I removed the IR filter in my SV305 a few years ago.  Unfortunately, it was glued in and removing required breaking the filter.

There are a lot of variations in IR filters.  DSLR IR filters usually cut off just before 650nm, just before Hydrogen Alpha, which is why people have to astromod their DSLRs.  Astronomy IR filters usually go up to ~700nm so HA is not removed.

Also, newer camera sensors are considerably more sensitive to IR than even 5 years ago.

**Comment by gigaklatch** [comment]
Oh, I had not even considered that. Sound like a cool route to go down.

ZWO have have IR cut protective windows with 21, 25 and 60 mm diameter. There is a thread on cloudy nights, where a guy opens up a SV705C to remove dust on the inside of the protective window. Doing some measurements in GIMP on the images of the inside of the camera posted in that thread, it looks like the window is more like 36 mm. So maybe I could get a regular unmounted  36mm luminance filter? They usualle cut away away wavelengths outside of a 400 to 700 nm range. Quite costly, but might be worth it if it does the trick. MIght be a bit thick though? Baader's have 2 mm thickness,  but perhaps there is room for that.

I would try and contact SVBony, but I have horrible experiences from dealing with them in the past, so it feels unlikely they could be of any help at all.

**Comment by gigaklatch** [comment]
At Aliexpress there seem to be available some round 36 mm(also 32.5mm if that would be a better fit)IR cut filters, they are stated to cut at 650 nm. Seems like the blue and green pixels of the IMX585 starts to increase in sensitivity again at about that wavelength, so might be good to cut there.

The thickness of these are not clearly known

HOw to know if these are of any kind of decent quality?

**Reply by aaronwmorris** [reply]
You will probably have to remove the lid from your SV705c to see what shape and size you need.

To be fair, you probably do not need to worry about quality so much.  In traditional astrophotography, you want a "flat" filter (eg 1/10 wave or better) to reduce distortions, but with all sky cameras, the lens causes so much distortion, you would probably never notice a bad filter anyway.

**Comment by gigaklatch** [comment]
I have not been able to remove the top of the camera , it's stuck like crazy. I have a filter that, circular and about 30 mm in diameter, I could replace the AR window with, but alas...

Instead I have now found a small 13x12 mm IR cut filter, only 0.21mm in thickness, which I have 3D printed a holder for that sits on the edges around the AR glass window, hovering just above it. The filter and holder is small enough where I can still screw on the lens onto the camera, with the filter and holder sandwiched between the AR glass window and the "output" of the lens.

**Comment by gigaklatch** [comment]
So, IR/UV cut filter is now installed. I was stupid enough to change two things at a time though, not only did I install the filter, but I also opened the aperture fully on the objective of my camera, thinking that the filter would cut out a lot of light and that I would open the aperture to gather some more of them precious photons.

But, without changing any software settings. It went from this:

![NOIRCUT](https://github.com/user-attachments/assets/bbe15e40-dbd4-446b-95b6-2697ddca0baa)

To this:
![IRCUT](https://github.com/user-attachments/assets/494ddbc9-8531-4d06-abb6-0f1b6707a137)

I do need to tweak the settings for the contrast enhancement as well as the stretching to attain some decent new images to compare.

And then, wait for the aurora to strike.

**Reply by jimmauck** [reply]
Do you have night exposure max set to 30? I wonder if it needs more time to dynamically adjust (increase) the exposure for target ADU.

**Reply by aaronwmorris** [reply]
That is a bit surprising.  Do you know the specs of the IR/Cut filter you installed?  Some filters are photographic and cutoff at ~650nm.  Astrophotography ircut filters usually go up to 700nm so they include Hydrogen Alpha.

Not that HA is necessarily important to an allsky camera, but some things related to aurora might be slightly impacted.  Some things like the SAR arc is ~630nm which might be impacted on a particularly bad filter.

I have also long suspected that ircut filters impact visibility of Airglow.

**Reply by gigaklatch** [reply]
The max is set to 30 seconds. The ADU is nowhere near the target. It seems like it ended up at around 7 during the most clear moments, and the target is 75.

Here you should be able to see the product page on Aliexpress. And well, they actually do seem to cut off really early at around 650 nm.
[https://www.aliexpress.com/item/1005003483323868.html?spm=a2g0o.productlist.main.1.7b737tan7tanz2&algo_pvid=c13c8f6c-e29d-4bcf-8b12-dfe2a953f6ce&algo_exp_id=c13c8f6c-e29d-4bcf-8b12-dfe2a953f6ce-0&pdp_ext_f=%7B%22order%22%3A%2212%22%2C%22eval%22%3A%221%22%7D&pdp_npi=4%40dis%21SEK%21120.20%2196.12%21%21%2110.88%218.70%21%40%2112000035251234196%21sea%21SE%214220988501%21X&curPageLogUid=GBdfmnqAKTMx&utparam-url=scene%3Asearch%7Cquery_from%3A](url)

Seems like I should keep looking for some filters more suitable for astrophotography.

**Comment by gigaklatch** [comment]
![2025-03-02 (1)](https://github.com/user-attachments/assets/5d458362-61cc-4370-a8e4-fa2604f1b4c7)

**Reply by aaronwmorris** [reply]
That is interesting info.

Keep in mind, there are practically no case studies for how a 650nm vs 700nm ircut filter will affect an allsky camera.  I have no idea what the impact will be.

**Comment by gigaklatch** [comment]
The filter I purchased a few months ago, with the intention of replacing the current AR glass window of the camera with, has a graph for the transmissivity for the various wavelengths of lights that look like this:

![2025-03-02 (5)](https://github.com/user-attachments/assets/263ea6f1-b4a3-4e49-83c9-8ad9241a50e9)

Like I said, I have been unable to get inside of the camera thus far. And the filter is a few tenths of a millimeter to large in diameter to drop on top of the AR window. Because of that it does not sit parallell with the AR window(and the camera sensor), and the camera lens is sure to run against and break the UV/IR filter.

But, the same seller offers one that is a few millimeters less in diameter, and it will sit perfectly.

![F9198J-SV705C--1200--5--compressed-docsmall com](https://github.com/user-attachments/assets/0da220b7-6d6b-49a2-af45-12a529c75369)

The filter will lay against the surfaces encircled in red. This is the same surface that my current solution, a small filter in a 3D-printed cradle, rests against.

Would there be a problem that there is an ever so small air gap between the AR window and the UV IR filter?

Should I be trying even harder to get inside the camera do the swap instead? So that there is only one piece of glass after light has exited the lens? Simply dropping a filter on top leaves you with less of a hassle whenever I would wish to restore the camera to its initial state.

**Reply by aaronwmorris** [reply]
I do NOT think the extra glass in the path would have a meaningful impact on the signal levels in an allsky system.  But I do not have any data to back that up.

I recall it was kind of a pain to remove the IR/Cut filter on my SV305.  I had to remove everything internally from the rear of the camera.  I also completely busted the filter because it was glued in.

**Comment by gigaklatch** [comment]
Had a short burst of auroral activity last night. The colors really do seem a lot better. You can discern the green band along the bottom,  with the larger red veil on top as well as some purple/blue columns to the left.

I hope that with some tweaking I can give everything a more "inky black" look, as for instance the sky looks a bit too grey, at least for my taste.
![ccd1_20250319_211200](https://github.com/user-attachments/assets/1d455e20-baff-48e3-83be-36eb4d8d4772)
![ccd1_20250319_211924](https://github.com/user-attachments/assets/2dbc0444-88e9-43be-b36b-ae0b0be657ce)

In a few weeks I will have a new filter to try out, the one that allows wavelengths closer to IR as well. I am not sure that will be better for the auroras, as it will allow in more light for the blue and green pixels, as ther respective filters start allowing in wavelengths again closer to the IR-range. Might become saturated and tend towards full white again.

Also have a few more lenses to try out. They are faster f1.2 ( 2.5 and 2.8 mm) vs the f1.6 (2.5 mm) of my current lens, though they cost only a quarter as much, so I guess that the might be worse in a lot of ways.

**Reply by aaronwmorris** [reply]
Those colors are very nice.

As for getting a blacker background.  I would try the MTF stretch function and start with a shadow cutoff of 0.03.  Slowly increase the shadow cutoff for a darker background.

**Comment by gigaklatch** [comment]
Some more activity tonight, had save FITS enabled. 

This is with my old settings:
![sd1](https://github.com/user-attachments/assets/623d0663-c901-434e-99ae-54d5f60ccd8e)
And now with MTF stretch function( and a tweak to the contrast enhancement):
![sd2](https://github.com/user-attachments/assets/c58c4283-c878-4a0f-a12f-b4daf7f62d09)

It's getting to where I would like it to be... I am considering a longer focal length lens, aiming the camera more to the north to maybe be able to catch some more details. But I have been unable to find suitable lenses for the rather large IMX 585 sensor in my camera. Perhaps the Meike 3.5mm could do the trick, but at f2.8 would it be too slow? Or how might that turn out?

**Reply by aaronwmorris** [reply]
Very nice.  Do you increase the saturation at all?

f/2.8 will not be too bad.  The f-ratio alone is not very meaningful when you change the sensor and the lens.  The aperture is considerably larger on the Meike lens than most CS-mount allsky lenses.  I have an f/7 lens I am working on with an APS-C camera and it can still work.

**Reply by gigaklatch** [reply]
I have the saturation bumped to 1.2. Higher than that and the stars starts looking silly, particularly the blue ones.

I was thinking of not changing the sensor(camera). If the f ratio is the fraction between focal length and aperture so that the focal length divided by the f ratio number gives you the aperture, my current lens(2.5/1.6= **1.5625**) has a larger aperture than the Meike does(3.5/2.8=**1.25**). But I guess  there would be other benefits to find with the Meike. It is designed for a much larger sensor(18x13.5mm), and so, for the smaller portion of the rendered image that the sensor on my camera would be capturing(11.2x6.3mm), I assume that the image would be well corrected, sharp and not have much distortion.

I know that there is an adapter made by ZWO that allows for attaching MFT lenses to some ZWO cameras( ones with a backfocus that is small enough). I am not sure that this adapter will fit on my SVBony camera. So I would have to cobble something together, 3D-printing and such, and that might turn out good enough... or not. Otherwise, I will have to give up the idea of using that lens with my camera.

I am now also looking for some fast C or CS mount lenses with around 4 or 5 mm of focal length that would render large enough image circles to fill the entirety of the IMX585 sensor. Most lenses either are for smaller sensors or are extremely expensive.

**Comment by gigaklatch** [comment]
With my current 2.5mm lens on my camera with an IMX 585 sensor I get an image, horizon to horizon, in all directions, and this perfectly fills the image sensor in the vertical direction, the sensor being 16:9 format there's naturally large parts of the sensor not illuminated to the sides.

Would a 5 mm lens basically give a 90 degree FOV on the vertical and a 160(=16 x 90/9) degree horizontal FOV? Or is it more complicated than that?

**Reply by aaronwmorris** [reply]
The focal length of a lens is not directly related to the FoV.  Obviously, there is a correlation, but there is no mathematical formula for it.

Much of this has to do with the design of the lens.  The size of the lenses, thickness, curvature, aperture, field stop, etc.

Generally, a larger focal length will result in a larger image circle.  The larger image circle just restricts the size of the sensor that can see the full image circle.  However, there are a couple of lenses in my Camera Simulator that deviate from this.

**Reply by gigaklatch** [reply]
Hmm, it seems like I might be using the wrong terms or something.

What I am thinking of maybe only applies to really long focal length gear, such as telescopes, where a doubling of the focal length, 2F, leads to the same camera capturing a patch of sky which has an area a quarter of that when instead using a telescope with focal length  F. Or at least, that what I had gathered from tinkering around with tools such as [https://astronomy.tools/calculators/field_of_view/](url). Four times the pixels for the same area of sky.

Is this just an approximation that works pretty well for really long focal length, where the light path is not really changed that much, whereas with fish eye lenses and stuff, all that goes out of the window?

I was not asking about the size of the image circle really. Just assume that the image circle covers the entirety of the sensor in both cases.

**Reply by aaronwmorris** [reply]
I completely understand what you are mentioning and you are absolutely right about the calculations.  Things are generally more simple with telescopes with longer focal lengths.  The challenge is the math only makes sense for a fisheye lens at the very center of the FoV.  While all telescopes have some form of distortion (barrel or pincushion) they are extremely "flat" compared to a fish-eye lens which has severe distortions.  We use them because they have these distortions.  :-)

Different lenses exhibit these distortions in different ways.  I would just look at the Camera Simulator in indi-allsky.  The math is just not as clear on some of these lenses.  There are lenses with different focal lengths, similar FoV and similar sized image circles.  This can only be true if the output of these lenses is distorted in different ways.

The arc seconds per pixel math just appears to break down under 10mm of focal length.
